extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -26,14 +28,88 @@ class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		$dependencies = array(
 			'lodash',
 			'wp-i18n',
 			'wp-token-list',
 		);
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+		$srcset_step         = 300;
+
+		if ( function_exists( 'jetpack_photon_url' )
+			&& preg_match_all( '/<img [^>]+>/', $content, $images )
+		) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					$orig_width  = $img_width[1];
+					$orig_height = $img_height[1];
```

Comment:
Should we cast integers here just in case, because PHP?

Reply:
e37e8473bc236f0997c4e85ebed59a4d43e53e21